src/app/dev/DevToys.Tools/Helpers/ListCompareHelper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Markdig.Helpers;
+using Microsoft.Extensions.Logging;
+
+namespace DevToys.Tools.Helpers;
+
+internal static partial class ListCompareHelper
+{
+    public static ResultInfo<string> Compare(
+        string firstList,
+        string secondList,
+        bool caseSensitive,
+        ListComparisonMode comparisonMode,
+        ILogger logger)
+    {
+        string[] separators = { "\r\n", "\n", "\r" };
+        ResultInfo<string> compareResult;
+        var listA = new List<string>(firstList.Split(separators, StringSplitOptions.RemoveEmptyEntries));
+        var listB = new List<string>(secondList.Split(separators, StringSplitOptions.RemoveEmptyEntries));
+        StringComparer stringComparer = StringComparer.CurrentCultureIgnoreCase;
+
+        if (caseSensitive)
+        {
+            stringComparer = StringComparer.CurrentCulture;
+        }
+
+        IEnumerable<string> listCompared = comparisonMode switch
+        {
+            ListComparisonMode.AInterB => GetAInterB(listA, listB, stringComparer),
+            ListComparisonMode.AOnly => GetAOnly(listA, listB, stringComparer),
+            ListComparisonMode.BOnly => GetBOnly(listA, listB, stringComparer),
+            _ => throw new NotSupportedException(),
+        };
+        compareResult = new(string.Join(Environment.NewLine, listCompared), true);
+        try
+        {
+            return compareResult;
```

Comment:
I do not expect this line to throw an exception. Should the `try` be higher in the method?

Reply:
Actually, given the current implementation, I do not see any risk of failure except that line 17 and 18 may crash if `firstList` or `secondList` is null.

Reply:
Right, i move the "try" upper